specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -351,6 +852,261 @@
       },
       "description": "Properties of network security perimeter."
     },
+    "NspProfile": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NspProfileProperties",
+          "description": "Properties of the network security perimeter profile"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the profile resource that is unique within a perimeter. This name can be used to access the resource."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Identifier of the network security perimeter profile in ARM id format."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
```

Comment:
>Resource [](http://example.com/codeflow?start=47&length=8)

Why do tou need name id and type if it is already defied in the resource. Also please note that the resource is currently defined with location. That would make this a tracked resource. Is that your intention ?

Reply:
For reference : [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/preview/2022-02-01-preview/network.json#L140](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/preview/2022-02-01-preview/network.json#L140)

Reply:
This resource definition is common for almost all the network resources. We had taken the copy from the stable. We need to respond with the location the user supplied while creating the response. We need this to be a tracked resource too. We had used the default routing type too in the manifest file.

Reply:
I mean you have already specified that the object has all the properties from the Resource Object. This addition of name type and id is a duplication of the resource definition from there source version